### ISSUE TITLE
Retrieval of Chart List Results in Failure in Large Clusters

### ISSUE THREAD

**Issue by toddtee** [komodorio/helm-dashboard#182]
### Description

Hi, thanks for this project, I can see alot of value in it!

Our organisation has close to 1000 namespaces in some clusters, and the following command fails retrieving the list of charts:
```level=warning msg="Failed command: [helm ls --all --output json --time-format 2006-01-02T15:04:05Z07:00 --all-namespaces]"```

We suspect the list of items it has to return results in a timeout or similar.

Thanks again!

### Screenshots

![helm-ls-fail](https://user-images.githubusercontent.com/15132637/211708538-6ad0dccc-1198-4b6b-aaab-bbc57b7996de.png)

**Comment by hdm23061993** [comment]
Can you check the execution time for the same command from cli and see how long does it take?
`time helm ls --all --output json --time-format 2006-01-02T15:04:05Z07:00 --all-namespaces`

**Comment by toddtee** [comment]
Hi @hdm23061993, I just ran this locally against the cluster I was trying to utilise `helm-dahsboard` with, and the timing results are as follows:
```________________________________________________________
Executed in   65.79 secs    fish           external
   usr time   29.01 secs    0.09 millis   29.01 secs
   sys time    5.25 secs    2.04 millis    5.25 secs```

**Comment by toddtee** [comment]
Upon further investigation, turns out that the `helm ls` command was being killed. It was a bit confusing as the container wasn't OOMing, and the pods weren't restarting though.

However, upon viewing the kernel logs on the worker nodes, we noticed that indeed the `cgroup` memory limits were being exceeded and the `helm` processes was being reaped.

We increased memory limits for the dashboard by a fair amount, and this has fixed the issue.

TL;DR
Pod needed much more memory allocated.

Happy to close if everyone else is.

**Comment by hdm23061993** [comment]
Yup, the issue can be closed
Thanks 😊

**Comment by toddtee** [comment]
@undera @hdm23061993 

Just a follow up; I still think there may be a retrival time issue. We still get that red popup "Failed to get list of charts". But this time, the process is still not killed (now that it has more memory). I then hit refresh on the browser, and the list of releases/namespaces shows up.

Again, might or may not be related to the time it takes to receive the `helm ls` result. Worth an issue? More of a UI improvement.

**Comment by undera** [comment]
I should say the scale you mention is quite far from what I had in mind while creating the helm-dashboard :) We'll need to rethink quite a bit the way we retrieve and present data in such cases

**Comment by hdm23061993** [comment]
Pagination is the way to go in my opinion
